src/integrationtests/java/com/aws/greengrass/integrationtests/logmanager/LogManagerTest.java

```diff
@@ -350,4 +353,105 @@ void GIVEN_log_manager_in_errored_state_WHEN_restarted_THEN_logs_upload_is_reatt
+        List<File> allFiles = new ArrayList<>();
+        File[] files = folder.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (file.isFile()
+                        && logFileNamePattern.matcher(file.getName()).find()
+                        && file.length() > 0) {
+                    allFiles.add(file);
+                }
+            }
+        }
+        assertEquals(1, allFiles.size());
+        logManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.set(false);
+    }
+
+    //TODO: this test is only for getting some certain level of knowledge of current change uploading active log file
+    // . It will be eventually removed.
+    @Test
+    void GIVEN_system_config_with_small_periodic_interval_WHEN_active_logs_included_THEN_logs_are_uploaded_to_cloud(
+            ExtensionContext ec) throws Exception {
+        when(cloudWatchLogsClient.putLogEvents(any(PutLogEventsRequest.class)))
+                .thenReturn(PutLogEventsResponse.builder().nextSequenceToken("nextToken").build());
+        logManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.set(true);
+        ignoreExceptionOfType(ec, NoSuchFileException.class);
+        LogManager.getRootLogConfiguration().setStoreDirectory(tempRootDir);
+        tempDirectoryPath = LogManager.getRootLogConfiguration().getStoreDirectory().resolve("logs");
+        String fileName = LogManager.getRootLogConfiguration().getFileName();
+        Files.createDirectory(tempDirectoryPath);
+        for (int i = 0; i < 5; i++) {
+            createTempFileAndWriteData(tempDirectoryPath, fileName, ".log");
+        }
+        createTempFileAndWriteData(tempDirectoryPath, fileName + ".log", "");
+
+        setupKernel(tempRootDir, "smallPeriodicIntervalSystemComponentConfig.yaml");
+        //TOdo: a better mechanism will be written
+        TimeUnit.SECONDS.sleep(30);
+        verify(cloudWatchLogsClient, atLeastOnce()).putLogEvents(captor.capture());
+
+        List<PutLogEventsRequest> putLogEventsRequests = captor.getAllValues();
+        assertEquals(1, putLogEventsRequests.size());
+
+        for (PutLogEventsRequest request : putLogEventsRequests) {
+            assertEquals(calculateLogStreamName(THING_NAME, LOCAL_DEPLOYMENT_GROUP_NAME), request.logStreamName());
+            assertEquals("/aws/greengrass/GreengrassSystemComponent/" + AWS_REGION + "/System",
+                    request.logGroupName());
+            assertNotNull(request.logEvents());
+            assertTrue(request.logEvents().size() >= DEFAULT_LOG_LINE_IN_FILE * 6);
+            assertTrue(request.logEvents().stream().mapToLong(value -> value.message().length()).sum()
+                    >= DEFAULT_FILE_SIZE * 6);
+        }
+        File folder = tempDirectoryPath.toFile();
+        Pattern logFileNamePattern = Pattern.compile(String.format(DEFAULT_FILE_REGEX, fileName));
+        List<File> allFiles = new ArrayList<>();
+        File[] files = folder.listFiles();
```

Comment:
There is a lot of overlap between this test and the one above it is really hard to understand what this is testing.  Try using better naming